Generate and push metrics nightly with GHA (SOFTWARE-4419)

edquist opened the pull request:
Not clear yet if this is supposed to publish to the gh-pages branch of the current repo, or of a separate `metrics` repo.

SOFTWARE-4419 seems to suggest the latter, which would require a second repo checkout, and in that case it's not so clear to me if there is anything special that will need for the `ad-m/github-push-action@master` action to work right...

edquist committed:
WIP - initial sketch of nightly GHA metrics gen & push (SOFTWARE-4419)

brianhlin commented on .github/workflows/metrics.yml:
```diff
@@ -0,0 +1,43 @@
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false  # use personal token, not GITHUB_TOKEN
```
What's this for? Doesn't `GITHUB_TOKEN` have repo write access?

edquist replied:
I am not sure if we need it.  Do we have an example somewhere that does a git push to github?

I was starting with the example i found here:
https://github.com/marketplace/actions/github-push

edquist replied:
Dropping this in favor of doing a push similar to

https://github.com/path-cc/path-cc.github.io/blob/master/.github/workflows/web-preview.yml

since we're pushing to a separate repo now

brianhlin commented on .github/workflows/metrics.yml:
```diff
@@ -0,0 +1,43 @@
+        git commit -m "nightly metrics update"
+
+    - name: Push changes
+      uses: ad-m/github-push-action@master
```
I'm not keen on following `master`. Let's use the latest version `v0.6.0`

edquist replied:
I switched to v0.6.0, and then dropped the ad-m/github-push-action altogether, since we're now pushing to a separate repo.

edquist commented on .github/workflows/metrics.yml:
```diff
@@ -0,0 +1,43 @@
+        git commit -m "nightly metrics update"
+
+    - name: Push changes
+      uses: ad-m/github-push-action@master
```
```suggestion
      uses: ad-m/github-push-action@v0.6.0
```

edquist committed:
use fixed github-push-action version (SOFTWARE-4419)

edquist committed:
push results to separate path-cc/metrics repo (SOFTWARE-4419)

ssh-{keyscan,agent,add} setup taken from:

https://github.com/path-cc/path-cc.github.io/blob/master/.github/workflows/web-preview.yml

which itself credits its SSH key recipe from:

https://www.webfactory.de/blog/use-ssh-key-for-private-repositories-in-github-actions

edquist committed:
rename deploy key secret (SOFTWARE-4419)

edquist commented:
@brianhlin, are you able to set up an appropriate `secrets.METRICS_DEPLOY_KEY` ?

Or if it's something for me to do let me know...

brianhlin commented on .github/workflows/metrics.yml:
```diff
@@ -0,0 +1,46 @@
+        git config --local user.name "Automatic preview publish"
+        git add campuses-with-active-researchers.csv
+        git add campus-contributions.json
+        git commit -m "nightly metrics update"
```
```suggestion
        git commit -m "nightly metrics update (from ${GITHUB_REPOSITORY} ${GITHUB_WORKFLOW} ${GITHUB_RUN_ID}.${GITHUB_RUN_NUMBER})"
```

For more details using https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables

brianhlin reviewed (approved):
One suggestion, otherwise LGTM

brianhlin commented:
Deploy key added to this repo and https://github.com/path-cc/metrics/. Have you tested this to ensure that it works? Perhaps adding a trigger for the workflow upon push to `master` that we revert in a subsequent PR?

edquist committed:
use more verbose commit message

per blin

Co-authored-by: Brian Lin <brianhlin@gmail.com>

edquist committed:
add temporary on: push for metrics run (SOFTWARE-4419)

edquist commented:
I certainly have not tested it as I don't know of a way to  :)

But yeah i went ahead and added a temporary `on: push` as you are suggesting
